src/services/inlayHints.ts

```diff
@@ -423,6 +451,308 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
         });
     }
 
+    function typeToInlayHintParts(type: Type): InlayHintDisplayPart[] | string {
+        if (!shouldUseInteractiveInlayHints(preferences)) {
+            return printTypeInSingleLine(type);
+        }
+
+        // TODO: No truncation?

```

Comment:
Marked as a `TODO` since I'm not sure of how we want to deal with truncation here. A lazy solution would be to transform the display parts into a string and then check its length. If it's above the limit, then I return the truncated string instead.

Reply:
Resolving this as we've decided to not have a truncation limit for now and see what happens.